demos/custom-dialogs/custom-dialog.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Blockly Demos: Custom Dialogs
+ *
+ * Copyright 2016 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CustomDialog = {}
+
+/** Override Blockly.alert() with custom implementation. */
+Blockly.alert = function(message, callback) {
+    console.log('Alert: ' + message);
+    CustomDialog.show('Alert', message, {
+        onCancel: callback
+    });
+};
+
+/** Override Blockly.confirm() with custom implementation. */
+Blockly.confirm = function(message, callback) {
+    console.log('Confirm: ' + message);
+    CustomDialog.show('Confirm', message, {
+        showOkay: true,
+        onOkay: function() {
+            callback(true)
+        },
+        showCancel: true,
+        onCancel: function() {
+            callback(false)
+        }
+    });
+};
+
+/** Override Blockly.prompt() with custom implementation. */
+Blockly.prompt = function(message, defaultValue, callback) {
+    console.log('Prompt: ' + message);
+    CustomDialog.show('Prompt', message, {
+        showInput: true,
+        showOkay: true,
+        onOkay: function() {
+            callback(CustomDialog.inputField.value)
+        },
+        showCancel: true,
+        onCancel: function() {
+            callback(null)
+        }
+    });
+    CustomDialog.inputField.value = defaultValue;
+};
+
+
+CustomDialog.hide = function() {
+    if (CustomDialog.backdropDiv_) {
+        CustomDialog.backdropDiv_.style.display = 'none'
+        CustomDialog.dialogDiv_.style.display = 'none'
+    }
+}
+
+/**
+ * Shows the dialog.
+ * Allowed options:
+ *  - showOkay: Whether to show the OK button.
+ *  - showCancel: Whether to show the Cancel button.
+ *  - showInput: Whether to show the text input field.
+ *  - onOkay: Callback to handle the okay button.
+ *  - onCancel: Callback to handle the cancel button.
+ */
+CustomDialog.show = function(title, message, options) {
+    var backdropDiv = CustomDialog.backdropDiv_;
+    var dialogDiv = CustomDialog.dialogDiv_;
+    if (!dialogDiv) {
+        // Generate HTML
+        var body = document.getElementsByTagName('body')[0];
+        backdropDiv = document.createElement('div');
+        backdropDiv.setAttribute('id', 'backdrop');
+        backdropDiv.style.cssText =
+              'position: absolute;'
+              + 'top: 0px;'
```

Comment:
0, not 0px.
'+' needs to be at end of previous line to avoid undefined behaviour in JavaScript.